Automattic/pocket-casts-android
# Slumber Studios - Add promo feature item

## Description
This adds feature item for Slumber Studios promo on upsell/ upgrade layouts.

## Testing Instructions

Test.1 Local feature flag is enabled

1. Apply this [patch](https://github.com/Automattic/pocket-casts-android/files/14175168/patch.patch)
2. Install `debugProd` build variant
3. Login with a Plus account with less than 30 days expiry
4. Go to `Profile` -> `Account`
5. ✅  Notice that a feature item is present for Slumber Studios promo in the account upgrade layout
6. Tap on Slumber Studios text
7. ✅ Notice that Slumber Studios site is opened
8. Sign out
9. Go to `Podcasts` tab
10. Tap the folder lock icon
11. ✅ Notice that a feature item is present for Slumber Studios promo in the Plus feature card
12. Tap on Slumber Studios text
13. ✅ Notice that Slumber Studios site is opened

Test.2 Local feature flag is disabled

1. Go to `Profile` -> `Settings` -> `Beta features`
2. Disable `Slumber Studios Promo`
3. Restart the app
4. Login with a `Plus` account with less than 30 days expiry
5. Go to `Profile` -> `Account`
6. ✅ Notice that a feature item is not present for Slumber Studios promo in the account upgrade layout. Instead, the "undying gratitude" text is shown
8. Sign out
9. Go to `Podcasts` tab
10. Tap the folder lock icon
11. ✅ Notice that a feature item is not present for Slumber Studios promo in the feature card. Instead, the "undying gratitude" text is shown


## Screenshots or Screencast 

Enabled | Disabled
----|----
![Screenshot_20240206_113616](https://github.com/Automattic/pocket-casts-android/assets/1405144/a2e1aff8-32e5-49f2-8ef2-dabad6b3780f)| ![Screenshot_20240206_113640](https://github.com/Automattic/pocket-casts-android/assets/1405144/02c5374a-8776-458e-a297-86ab0c9447a5)

Enabled | Disabled
----|----
![Screenshot_20240206_113904](https://github.com/Automattic/pocket-casts-android/assets/1405144/3e72f3c5-0702-4c4a-9379-69b27b0b6c6a) | ![Screenshot_20240206_113728](https://github.com/Automattic/pocket-casts-android/assets/1405144/5ff3da5f-4496-459e-a82c-0901d0565f3d)



## Checklist
 
#### I have tested any UI changes...
